src/rgw/rgw_tracer.h

```diff
@@ -0,0 +1,77 @@
+#ifndef TRACER_WRAPPER_H
+#define TRACER_WRAPPER_H
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+#include <jaegertracing/Tracer.h>
+#include <yaml-cpp/yaml.h>
+#include "common/config.h"
+#include "common/ceph_context.h"
+
+
+namespace RGW_Tracing
+{
+    class TracerConfiguration : public md_config_obs_t {
+        private:
+            bool enabled = true;
+
+        public:
+            TracerConfiguration();
+            bool is_enabled() const;
+            const char** get_tracked_conf_keys() const override;
+            void handle_conf_change(const ConfigProxy& conf,
+                                    const std::set<std::string>& changed) override;
+    };
+
+    
+
+    class Span {
```

Comment:
please add comments on the Tracer and Span APIs in the header file

Reply:
done in 7482e24